### ISSUE TITLE
Gradle config does not exist after Android Studio update. Project fails to build.

### ISSUE THREAD

**Issue by apptrash** [google/ExoPlayer#8927]
After updating android studio, when building, the following message is shown:
```
Build file '/home/alex/Разработка/src/android-git/exoplayer/extensions/av1/build.gradle' line: 14

A problem occurred evaluating project ':exoplayer-extension-av1'.
> Could not read script '/home/alex/.gradle/daemon/exoplayer/common_library_config.gradle' as it does not exist.
```
I'm using ExoPlayer r2.12.1 

Here is a project structure (among others):
```
cgc.launcher/settings.gradle
exoplayer/
```
Here is `cgc.launcher/settings.gradle` content:
```
include ':app'

gradle.ext.exoplayerRoot = '../exoplayer'
gradle.ext.exoplayerModulePrefix = 'exoplayer-'
apply from: new File(gradle.ext.exoplayerRoot, 'core_settings.gradle')
```

Also, from `cgc.launcher/app/build.gradle`:
```
...
implementation project(':exoplayer-library')
implementation project(':exoplayer-extension-ffmpeg')
implementation project(':exoplayer-extension-rtmp')
implementation project(':exoplayer-extension-okhttp')
...
```

Here is information about Android Studio: 
```
Android Studio 4.2
Build #AI-202.7660.26.42.7322048, built on April 29, 2021
Runtime version: 11.0.8+0-b944-P17168821 amd64
VM: OpenJDK 64-Bit Server VM by N/A
Linux 5.11.0-16-generic
GC: G1 Young Generation, G1 Old Generation
Memory: 2048M
Cores: 4
Registry: external.system.auto.import.disabled=true
Non-Bundled Plugins: org.jetbrains.kotlin
Current Desktop: ubuntu:GNOME
```
I will provide more information if required. What can I do to fix the problem?

**Comment by apptrash** [comment]
Rolled back the version to 
```
Android Studio 4.1.3
Build #AI-201.8743.12.41.7199119, built on March 11, 2021
Runtime version: 1.8.0_242-release-1644-b3-6222593 amd64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
Linux 5.11.0-16-generic
GC: ParNew, ConcurrentMarkSweep
Memory: 2014M
Cores: 4
Registry: ide.new.welcome.screen.force=true, external.system.auto.import.disabled=true
Current Desktop: ubuntu:GNOME
```
Everything is OK on this version

**Comment by alex3d** [comment]
The problem is as of Android Studio 4.2 `gradle.ext.exoplayerRoot` should be set to an absolute path. This probably could be mentioned in exoplayer readme.

Android Studio 4.2 is bundled with JDK 11 instead of JDK 8 in previous versions, and see https://github.com/gradle/gradle/issues/8286#issuecomment-456940831

**Comment by Samrobbo** [comment]
Hi @apptrash, it looks like `alex3d` has given you a possible solution, did it work? 

You could also try updating to the latest ExoPlayer release and see what effect that has.

**Comment by apptrash** [comment]
@Samrobbo, hi! Yeah, it does help. But now everyone who involved in project should set this path manually

**Comment by Samrobbo** [comment]
There may be an alternate way to do this, but I don't think it's specific to ExoPlayer. You might find some success asking a similar question on StackOverflow or in a gradle forum.

**Comment by apptrash** [comment]
Okay, I'll ask a similar question and write here what I found out.

**Comment by apptrash** [comment]
"${rootProject.projectDir}/../exoplayer" works fine for me

**Comment by S0und** [comment]
To be more specific
in your `settings.gradle`

Change this:
`gradle.ext.exoplayerRoot = '$rootDir/../../ExoPlayer'`

to this: 

`gradle.ext.exoplayerRoot = "${rootProject.projectDir}/../ExoPlayer"`

Obviously the initial path can be different, but the end of the day it has to correctly point to `common_library_config.gradle` which is in the root of the `/ExoPlayer/` folder

**Comment by icbaker** [comment]
After changing the README instructions above to require an absolute path, I took another look to see if we could remove that requirement.

I worked out the problem comes when constructing a `File` object from the `exoplayerRoot` string. Gradle says we can't call `new File` with a relative path, we have to use their `Project.file` method:
https://docs.gradle.org/current/userguide/working_with_files.html#sec:single_file_paths

And indeed I can use that to fix the problem by editing [`core_settings.gradle`](https://github.com/google/ExoPlayer/blob/release-v2/core_settings.gradle).

I've sent a change to make this fix - it will appear below when it's merged and should be included in 2.14.1 (but if people are depending on ExoPlayer locally they can of course apply the patch directly).
